File: server/src/main/java/com/campusmarketplace/server/service/NotificationService.java
package com.campusmarketplace.server.service;

import com.campusmarketplace.server.dto.response.NotificationResponse;

import java.util.List;

public interface NotificationService {

    // Create a new notification
    void createNotification(
            String recipientEmail,
            String type,
            String title,
            String message,
            String taskId
    );

    // Get notifications of currently logged-in user
    List<NotificationResponse> getMyNotifications();

    // Get unread notification count
    long getUnreadCount();

    // Mark a single notification as read
    NotificationResponse markAsRead(String notificationId);

    // Mark all notifications as read
    void markAllAsRead();
}

File: client/src/services/NotificationService.js
import api from "../utils/axiosConfig";

const NotificationService = {

  // =====================================================
  // GET ALL NOTIFICATIONS OF LOGGED-IN USER
  // =====================================================

  async getMyNotifications() {
    const response = await api.get(
      "/notifications"
    );

    return response.data;
  },

  // =====================================================
  // GET UNREAD NOTIFICATION COUNT
  // =====================================================

  async getUnreadCount() {
    const response = await api.get(
      "/notifications/unread-count"
    );

    // Backend response:
    // {
    //   "count": 3
    // }

    return response.data.count;
  },

  // =====================================================
  // MARK ONE NOTIFICATION AS READ
  // =====================================================

  async markAsRead(notificationId) {
    const response = await api.put(
      `/notifications/${notificationId}/read`
    );

    return response.data;
  },

  // =====================================================
  // MARK ALL NOTIFICATIONS AS READ
  // =====================================================

  async markAllAsRead() {
    const response = await api.put(
      "/notifications/read-all"
    );

    return response.data;
  },
};

export default NotificationService;